ofrak_core/ofrak/core/free_space.py

```diff
@@ -370,15 +358,50 @@ async def _get_partially_overlapping_resources(
         )
 
 
+async def _find_and_delete_overlapping_children(resource: Resource, freed_range: Range):
+    # Note this filter calculation has the potential to be very expensive if, for instance,
+    # the resource is an entire program segment...
+    overlap_resources = list(
+        await resource.get_children_as_view(
+            MemoryRegion,
+            r_filter=ResourceFilter(
+                tags=(MemoryRegion,),
+                attribute_filters=(
+                    ResourceAttributeRangeFilter(MemoryRegion.VirtualAddress, max=freed_range.end),
+                ),
+            ),
+        )
+    )
+    for possible_overlapping_child in overlap_resources:
+        if (
+            possible_overlapping_child.virtual_address > freed_range.start
+            or possible_overlapping_child.end_vaddr() > freed_range.start
+        ):
+            await possible_overlapping_child.resource.delete()
+            await possible_overlapping_child.resource.save()
+
+
+def _get_fill(freed_range: Range, fill: Optional[bytes]):
+    if not fill:
+        return b"\x00" * freed_range.length()
+    else:
+        diff_len = freed_range.length() - len(fill)
+        if diff_len < 0:
+            raise ValueError("config.fill value cannot be longer than the range to be freed.")
+        return fill + b"\x00" * diff_len
+
+
 @dataclass
 class FreeSpaceModifierConfig(ComponentConfig):
     """
     Configuration for modifier which marks some free space.
 
     :var permissions: memory permissions to give the created free space.
+    :var fill: bytes to fill the free space with
     """
 
     permissions: MemoryPermissions
+    fill: Optional[bytes] = None
 
 
 class FreeSpaceModifier(Modifier[FreeSpaceModifierConfig]):
```

Comment:
The Docstring (line 41) also should be updated, something like:

_...by replacing its data with fill or null bytes and tagging..._